scripts/apps/archive/controllers/file-upload-error-modal.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+
+import {gettext, gettextPlural} from 'core/utils';
+import Button from 'core/ui/components/Button';
+import {Modal} from 'core/ui/components/Modal/Modal';
+import {ModalHeader} from 'core/ui/components/Modal/ModalHeader';
+import {ModalBody} from 'core/ui/components/Modal/ModalBody';
+import {ModalFooter} from 'core/ui/components/Modal/ModalFooter';
+import {appConfig} from 'appConfig';
+
+interface IProps {
+    closeModal(): void;
+}
+
+interface IFile {
+    valid: boolean;
+    name: string;
+    width: number;
+    height: number;
+    type: string;
+}
+
+export function fileUploadErrorModal(
+    invalidFiles: Array<IFile>,
+) {
+    return class FileUploadErrorModal extends React.PureComponent<IProps> {
+        render() {
+            if (invalidFiles.length > 0) {
+                return (
+                    <Modal>
+                        <ModalHeader onClose={this.props.closeModal}>
+                            {
+                                gettextPlural(
+                                    invalidFiles.length,
+                                    'The file was not uploaded',
+                                    'Some files were not uploaded',
+                                )
+                            }
+                        </ModalHeader>
+                        <ModalBody>
+                            <h4>
+                                {
+                                    gettext(
+                                        'Minimum allowed image size is {{minWidth}}x{{minHeight}}',
```

Comment:
Looks like you are displaying the error about incorrect image size unconditionally for all files including non-image files. Could you explain this? When I'm asking a question, it doesn't mean that your solution is incorrect or needs changes. I just want to hear your reasoning behind the decision. This is the same question as these two https://github.com/superdesk/superdesk-client-core/pull/3375#discussion_r392095380 https://github.com/superdesk/superdesk-client-core/pull/3375#discussion_r392095380

Reply:
yeah, I'll fix this. But the idea is that I'm returning the error object in case of invalid images only. And in case of audio and video will simply show the name of the invalid files as we don't know what could be the error beforehand.

Reply:
> But the idea is that I'm returning the error object in case of invalid images only.

I understand that this "works", but it's a very unreliable solution in the long run because it's implicit and for a developer making future changes there is no way to know that it would cause a bug if an error object was returned if there's an error for a video file for example. Thanks for fixing this.